learn-data-structures-by-building-a-shopping-cart/index18.js

```diff
@@ -0,0 +1,65 @@
+const products = [
+  {
+    id: 0,
+    name: "Vanilla buttercream cupcake",
+    price: 2.99
+  },
+  {
+    id: 1,
+    name: "French Macaroon",
+    price: 3.99
+  },
+  {
+    id: 2,
+    name: "Fruit sprinkles cupcake",
+    price: 3.99
+  },
+  {
+    id: 3,
+    name: "Pink flower cupcake",
+    price: 5.99
+  }
+];
+
+class ShoppingCart {
+  constructor() {
+    this.items = [];
+    this.discountPercentage = 30;
+  }
+
+  addItem(id, products) {
+    const item = products.find(item => item.id === id);
+    this.items.push(item);
+  }
+
+  getItems() {
+    return this.items;
+  }
+  
+  getCount() {
+    return this.items.length;
+  }
+  
+  removeItem(id) {
+    const index = this.items.indexOf(id);
+    this.items.splice(index, 1);
+  }
+  
+  clearCart() {
+    this.items = [];
+  }
+  
+  applyDiscount(amount) {
+    return parseFloat((this.discountPercentage / 100 * amount).toFixed(2));
+  }
+}
+
+/*
+
+We should also handle tax.
```

Comment:
We should also handle sales tax.
Taxes vary from state to state, so let's use Texas's rate of 8.25%. 
Create a new variable called `taxRate` in the `constructor` function and assign 8.25 to it.